src/Azure.Functions.Cli/Common/AuthManager.cs

```diff
@@ -0,0 +1,406 @@
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_RUNTIME_VERSION", "1.0.0");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_TOKEN_STORE", "True");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_UNAUTHENTICATED_ACTION", "AllowAnonymous");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_HOSTNAME", hostName);
+
+            // Create signing and encryption keys for local testing
+            string encryptionKey = ComputeSha256Hash(clientSecret);
+            string signingKey = ComputeSha256Hash(clientId);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_ENCRYPTION_KEY", encryptionKey);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_SIGNING_KEY", signingKey);
+            MiddlewareAuthSettings.Commit();
+
+            ColoredConsole.WriteLine(Yellow($"Created {SecretsManager.MiddlewareAuthSettingsFileName} with authentication settings necessary for local development.\n" +
+                $"Running this function locally will only work with the Function Host default port of {StartHostAction.DefaultPort}"));
+
+            if (_workerRuntime == WorkerRuntime.dotnet)
+            {
+                // If this is a dotnet function, we also need to add this file to the .csproj 
+                // so that it copies to \bin\debug\netstandard2.x when the Function builds
+                string csProjFile = GetCSProjFilePath();
+
+                if (csProjFile == null)
+                {
+                    throw new CliException($"Auth settings file {SecretsManager.MiddlewareAuthSettingsFileName} could not be added to a .csproj and will not be present in the the bin or output directories.");
+                }
+
+                ModifyCSProj(csProjFile);
+            }
+        }
+
+        public static string GetCSProjFilePath()
+        {
+            // If we're in the Function root, the .csproj file will be in this folder
+            var csProjFiles = FileSystemHelpers.GetFiles(Environment.CurrentDirectory, searchPattern: "*.csproj").ToList();
+            if (csProjFiles.Count == 1)
+            {
+                return csProjFiles.First();
+            }
+
+            // If we're in the function root\bin\debug\netstandard2.x, the .csproj file will be up three directories
+            try
+            {
+                var functionDir = Path.GetFullPath(Path.Combine(Environment.CurrentDirectory, @"..\..\..\"));
+                var functionDirProjFiles = FileSystemHelpers.GetFiles(functionDir, searchPattern: "*.csproj").ToList();
+
+                if (functionDirProjFiles.Count == 1)
+                {
+                    return functionDirProjFiles.First();
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            catch (DirectoryNotFoundException e)
```

Comment:
Rather than relying on exceptions, it's best practice to explicitly check to see if the directory exists - e.g. `Directory.Exists(...)`.